api/src/main/java/org/openmrs/module/mirebalais/setup/ConfigurationSetup.java

```diff
@@ -0,0 +1,415 @@
+        }
+
+        // Setup Location Tags from PIH Config (TODO: Move this to use locations or locationtagmaps domains in Iniz)
+        setStatus("Configuring location tags from pih config");
+        LocationTagSetup.setupLocationTags(locationService, config);
+
+        // Setup primary identifier type metadata mappings (TODO: Move this to metadatamapping iniz domain with site-based exclusion filters)
+        setStatus("Configuring primary identifier type");
+        MetadataMappingsSetup.setupPrimaryIdentifierTypeBasedOnCountry(metadataMappingService, patientService, config);
+
+        // Setup identifier generators (TODO: This is a remaining configuration initiative piece.  Move to iniz and/or pih config)
+        setStatus("Configuring identifier generators");
+        PatientIdentifierSetup.setupIdentifierGeneratorsIfNecessary(identifierSourceService, locationService, config);
+
+        // TODO: Move the below to config.  All this is ultimately doing is ensuring the setting of a single global property:
+        //  registrationcore.identifierSourceId = <uuid of identifier source configured as autogeneration option for the emrapi primary identifier type>
+        setStatus("Configuring global properties for registration modules");
+        ModuleFactory.getStartedModuleById("registrationapp").getModuleActivator().started();
+
+        // Initialize PACS HL7 listener, if the PACS_INTEGRATION component is enabled in pih config
+        // TODO:  1. Determine if we can just enable this all the time, and if that will cause any adverse effects
+        // TODO:  2. Or determine if we can enable it all the time, but switch inside the listener on the component, to determine whether to act or not
+        // TODO:  3. If we leave this like this, determine if we can call this setup method over and over in a running server, or if we need to code around that
+        //           In testing, when running configure, I get this:  java.lang.RuntimeException: java.net.BindException: Address already in use (Bind failed)
+        setStatus("Configuring PACS HL7 Listener");
+        PacIntegrationSetup.setup(config);
+
+        // TODO: This seems harmless here for refreshing, but we should determine if
+        // TODO: 1. Has this run everywhere it needs to, and can we remove this code altogether now
+        // TODO: 2. If not, can we move this to a liquibase changeset instead
+        setStatus("Migrating attachment concepts if necessary");
+        AttachmentsSetup.migrateAttachmentsConceptsIfNecessary(conceptService);
+
+        // Configure global scripts and css files defined in the configuration directory
+        setStatus("Configuring global resources");
+        GlobalResourceSetup.includeGlobalResources();
+
+        // Rebuild search index (TODO: Confirm this is intended to run here, prior to metadata loading, rather than at the end of the process(
```

Comment:
Probably should happen at the end you are right...

Reply:
I wasn't sure.  For example if the only reason this is needed at all is because there might be direct SQL changes done in the Liquibase step, and we need to rebuild the search index to account for it, then it's fine here probably.  Also, if the metadata loading steps below depend on the index to actually find things like Concepts via the API, then it definitely needs to go before them.  I would think that when other things are loaded in via Iniz, using the API for all metadata updates, that they would update the search index natively via the API.

Reply:
FWIW, this is really just dev tooling, not something ever used in production.  It was added for the Haiti HIV test server, because we need to rerun the index after the migration is run to pick up all the patients.  I'm 85% sure we don't use it anywhere else.